src/runtime_src/core/pcie/driver/linux/xocl/subdev/xgq.c

```diff
@@ -2780,7 +2785,31 @@ static inline bool xgq_device_is_ready(struct xocl_xgq_vmr *xgq)
 				return true;
 		}
 	}
-	
+
+	return false;
+}
+
+static bool vmr_check_sc_is_ready(struct xocl_xgq_vmr *xgq)
+{
+	struct xgq_cmd_cq_vmr_payload *vmr_status =
+		(struct xgq_cmd_cq_vmr_payload *)&xgq->xgq_cq_payload;
+	int i, ret;
+
+	for (i = 0; i < MAX_SC_WAIT; i++) {
+		ret = vmr_status_query(xgq->xgq_pdev);
+		if (ret)
+			XGQ_ERR(xgq, "received error %d for vmr_status_query xgq request", ret);
+
+		if (vmr_status->sc_is_ready) {
+			XGQ_INFO(xgq, "SC is ready and active");
+			return true;
+		}
+
+		XGQ_WARN(xgq, "waiting for SC to be in ready state");
```

Comment:
It would be great if the time currently waited is in the message. 

How often this message is produced (e.g., once every 5 seconds) should be driven by a variable as well.